lms/djangoapps/grades/new/course_grade.py

```diff
@@ -70,27 +104,46 @@ def grade_value(self):
         self._log_event(log.warning, u"grade_value, percent: {0}, grade: {1}".format(percent, letter_grade))
         return grade_value
 
-    @property
-    def has_access_to_course(self):
+    @lazy
+    def chapter_grades(self):
         """
-        Returns whether the course structure as seen by the
-        given student is non-empty.
+        Returns a list of chapters, each containing its subsection grades,
+        display name, and url name.
         """
-        return len(self.course_structure) > 0
+        chapter_grades = []
+        for chapter_key in self.course_structure.get_children(self.course.location):
+            chapter = self.course_structure[chapter_key]
+            chapter_subsection_grades = []
+            children = self.course_structure.get_children(chapter_key)
+            for subsection_key in children:
+                chapter_subsection_grades.append(
+                    self.subsection_grade_factory.create(self.course_structure[subsection_key], read_only=True)
+                )
+
+            chapter_grades.append({
+                'display_name': block_metadata_utils.display_name_with_default_escaped(chapter),
+                'url_name': block_metadata_utils.url_name_for_block(chapter),
+                'sections': chapter_subsection_grades
+            })
+        return chapter_grades
 
     @property
     def percent(self):
         """
         Returns a rounded percent from the overall grade.
         """
-        return self._calc_percent(self.grade_value)
+        if self._percent is None:
```

Comment:
Those two still require an underlying internal variable, as they are assigned when a course grade is loaded from the database (my understanding is that you can't assign something that is `lazy`, only read it). That doesn't mean they can't be `lazy`, but they can't mimic `chapter_grades` directly, because `chapter_grades`'s value only can be assigned when the property is accessed.